spylib/store.py

```diff
@@ -163,10 +170,25 @@ async def execute_gql(self, query: str, variables: Dict[str, Any] = {}) -> Dict[
             errorlist = '\n'.join(
                 [err['message'] for err in jsondata['errors'] if 'message' in err]
             )
-            if THROTTLED_ERROR_MESSAGE in errorlist:
-                raise ShopifyThrottledError(
-                    f'Store {self.name}: The Shopify API token is throttling. '
+            error_code_list = '\n'.join(
+                [
+                    err['extensions']['code']
+                    for err in jsondata['errors']
+                    if 'extensions' in err and 'code' in err['extensions']
+                ]
+            )
+            if MAX_COST_EXCEEDED_ERROR_CODE in error_code_list:
+                raise ShopifyExceedingMaxCostError(
+                    f'Store {self.name}: The Shopify API token is throttling.'
```

Comment:
Question of English: Is it really "throttling" when Shopify will just plainly refuse the query cause it's too big???

Reply:
Good question, I'd say you're right because even if the pool was full the query would never run!